jni/src/dartjni.c

```diff
@@ -674,18 +678,36 @@ Java_com_github_dart_1lang_jni_PortProxy__1invoke(JNIEnv* env,
     c_post.value.as_array.length = sizeof(c_post_arr) / sizeof(c_post_arr[0]);
 
     Dart_PostCObject_DL(port, &c_post);
+
     while (!result->ready) {
       wait_for(&result->cond, &result->lock);
     }
+
     release_lock(&result->lock);
     destroy_lock(&result->lock);
     destroy_cond(&result->cond);
-    jobject object = result->object;
-    free(result);
-    return object;
   } else {
-    return ((jobject(*)(uint64_t, jobject, jobject))functionPtr)(
+    result->object = ((jobject(*)(uint64_t, jobject, jobject))functionPtr)(
         port, (*env)->NewGlobalRef(env, methodDescriptor),
         (*env)->NewGlobalRef(env, args));
   }
-}
\ No newline at end of file
+  attach_thread();
+  load_class_global_ref(&_c_Object, "java/lang/Object");
+  load_class_global_ref(&_c_Long, "java/lang/Long");
+  load_method(_c_Long, &_m_Long_init, "<init>", "(J)V");
+  jobject first = (*env)->NewObject(env, _c_Long, _m_Long_init, (jlong)result);
```

Comment:
If I understand correctly, we're boxing here. Why do we need boxing for Windows?

Reply:
We're returning an array of Objects so unfortunately, we have to box the long value.